File: python/leetcode/algorithms/easy/119-杨辉三角 II.py
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        lists = []
        for i in range(rowIndex + 1):
            temp = [1] * (i + 1)
            if i > 1:
                for j in range(1, i):
                    temp[j] = lists[i - 1][j - 1] + lists[i - 1][j]
            lists.append(temp)
        return lists[rowIndex]


s = Solution()
print(s.getRow(5))

File: go/leetcode/easy/119-杨辉三角 II.go
package main

func getRow(rowIndex int) []int {
	lists := [][]int{}
	for i := 0; i < rowIndex+1; i++ {
		temp := make([]int, i+1)
		for j := 0; j < len(temp); j++ {
			if j == 0 || j == len(temp)-1 {
				temp[j] = 1
			} else {
				temp[j] = lists[i-1][j-1] + lists[i-1][j]
			}
		}
		lists = append(lists, temp)
	}
	return lists[rowIndex]
}

func main() {

}
